src/views/ConnectionBrowser.ts

```diff
@@ -95,12 +95,12 @@ export function initializeConnectionBrowser(context: vscode.ExtensionContext) {
 
         if (newName) {
           try {
-            let index;
             // First rename the connection details
-            const connections = GlobalConfiguration.get<ConnectionData[]>(`connections`) || [];
-            index = connections.findIndex(connection => connection.name === server.name);
+            const existingConnection = ConnectionManager.getByName(server.name)!;
+            let { index, data } = existingConnection!
```

Comment:
Since we don't reuse `existingConnection`...
```suggestion
            let { index, data } = ConnectionManager.getByName(server.name)!;
```